### ISSUE TITLE
Hint actions without prompt

### ISSUE THREAD

**Issue by ilysym** [atlas-engineer/nyxt#2185]
**Is your feature request related to a problem? Please describe.**
Showing prompt window for every hint action is redundant and distracting in many cases, in particular for simple navigation.

**Describe the solution you'd like**
Add a flag argument to `query-hints` which will tell Nyxt to skip prompt and perform requested action once user have typed hint letters, without additional confirmation. This would allow user to define "quicker" versions of existing hint commands.

**Describe alternatives you've considered**
This proposal probably doesn't require alternatives because browser behaviour won't need to change with default configuration.

**Additional context**
In Vimium, luakit and most of similar software, hints actions are performed without additional prompt. I think this is most reasonable default behaviour for navigation actions.

When navigating many web sites, showing link URL doesn't provide any useful information, since links within the website often just a some constant URL + numeric id of a page. Moreover, a user doesn't typically care about link URL anyway and just wants to follow a link immediately.

On the other hand, being able to preview a link before following it is obviously useful in some cases. So it would be useful to have separate commands for these cases. Similar considerations are applicable for other hint actions (copy URL etc).

**Comment by aadcg** [comment]
Sounds reasonable to me. What Nyxt offers that others don't (AFAIK) is the ability to choose a link by both the hint string and the text associated with the link.

**Comment by Ambrevar** [comment]
Yup, we could totally add a command that shuts up the prompt, it's trivial

Any taker? :)

**Comment by aadcg** [comment]
I'll take it.

**Comment by aadcg** [comment]
@ambrevar, I expected that the following would exit from the prompt when there was a single item left. What am I missing?

```lisp
(prompt
 :auto-return-p t
 :history nil
 :sources
 (make-instance
  'hint-source
  :multi-selection-p t
  :constructor (lambda (source)
                 (declare (ignore source))
                 (add-element-hints :selector (if (current-mode 'web)
                                                  (hints-selector (current-mode 'web))
                                                  "a, button, input, textarea, details, select, img:not([alt=\"\"])")))))
```

**Comment by aadcg** [comment]
The following seems to be true - it's not possible to use the hinting mechanism without relying on the `prompt` interface, since it's not possible to exit from `(make-instance 'hint-source ...)` with a DOM element.

**Comment by aartaka** [comment]
Look at the code introduced in #2054, there is an example of interactive prompt programming. In particular, tests/renderer-online/set-url.lisp can be a good example here.

**Comment by aadcg** [comment]
I guess I see what you're trying to suggest, but it seems like it might not be a good idea to set a timeout to the prompt when all the hints have been computed. The user might be a slowpoke :D

**Comment by Ambrevar** [comment]
Is this solved then?  @aadcg

**Comment by aartaka** [comment]
@Ambrevar, I wouldn't say so -- the prompt is still visible and the hint-exact matching is not yet there. To enable this, we need to:
- `ffi-window-set-prompt-buffer-height` in the constructor of `hint-source`.
- Add a stronger `filter-postprocessor` to `hint-source`, so that the hint with matching hint letters is left as the sole suggestion, thus triggering `auto-return-p`.

@aadcg, you interested in completing those?

**Comment by aadcg** [comment]
@aartaka, I agree it's not solved too.  I'm interested, but I'll get to it later at some point.

**Comment by aartaka** [comment]
No problem :)
